transaction.go

```diff
@@ -43,6 +43,8 @@ type Transaction struct {
 	Customer                     *Customer                 `xml:"customer,omitempty"`
 	BillingAddress               *Address                  `xml:"billing,omitempty"`
 	ShippingAddress              *Address                  `xml:"shipping,omitempty"`
+	TaxAmount                    *NullDecimal              `xml:"tax-amount,omitempty"`
```

Comment:
I believe I understand and agree with the reasoning behind making this both a pointer and a NullDecimal, and I think it might be worth warning/explaining to future users/maintainers why it came to be that way.

In addition, maybe it's worth explaining that this is always populated when one of these structs is created from an API call of some kind, so users can potentially omit the nil check if they feel safe doing so, and go straight to checking for Valid.